### PR TITLE
Ali/worker bug fixes

### PR DESCRIPTION
**Pull request by mintona**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
Articles were not repopulating in old collections properly. Fixed the issue for now. Still needs optimization for the future. 

## Related Tickets & Documents
#30 

## Added tests?

- [x] yes
   * Added robustness to former tests for check that new articles would come in. 
   * Also fixed old tests that were not working b/c of an inefficient query
- [ ] no, because they aren't needed
- [ ] no, because I need help

## Added to documentation?

- [ ] docs.dev.to
- [ ] readme
- [ ] no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

File: app/controllers/reading_collections_controller.rb [hunk 1]
```diff
@@ -1,7 +1,9 @@
 class ReadingCollectionsController < ApplicationController
   def show
-    @collection = ReadingCollection.find_by(slug: params[:slug]).to_json
-    @articles = ReadingCollection.find_by(slug: params[:slug]).articles.to_json
+    reading_collection = ReadingCollection.find_by(slug: params[:slug])
+
+    @collection = reading_collection.to_json
+    @articles = reading_collection.articles.includes([:taggings]).to_json
```
Comment (mintona) [inline app/controllers/reading_collections_controller.rb:6]: Joel - the failing tests forced me to change this and include the `includes([:taggings])` in the query to get articles. I triple checked and it does not seem to be affecting the FE.

File: app/models/reading_collection.rb [hunk 2]
```diff
@@ -8,7 +8,7 @@ class ReadingCollection < ApplicationRecord
   acts_as_taggable_on :tags
 
   def get_articles
-    Article.where("created_at >= ?", 5.minutes.ago).
+    Article.where("created_at >= ?", 2.minutes.ago).
```
Comment (mintona) [inline app/models/reading_collection.rb:11]: For demo purposes

File: app/controllers/reading_collections_controller.rb (hunk 1 above)
Comment (philjdelong) [inline app/controllers/reading_collections_controller.rb:6]: Ah nice catch. I see where that could have been an issue

File: app/models/reading_collection.rb (hunk 2 above)
Comment (philjdelong) [reply app/models/reading_collection.rb:11]: perfect

File: app/workers/reading_collections/refresh_reading_collection_worker.rb [hunk 3]
```diff
@@ -5,10 +5,12 @@ class RefreshReadingCollectionWorker
     sidekiq_options queue: :high_priority, retry: 10
 
     def perform
-      reading_collections = ReadingCollection.where("updated_at < ?", 1.minute.ago)
+      reading_collections = ReadingCollection.where("updated_at < ?", 3.minutes.ago)
```
Comment (philjdelong) [inline app/workers/reading_collections/refresh_reading_collection_worker.rb:8]: Look\s good on time here too

File: app/controllers/reading_collections_controller.rb (hunk 1 above)
Comment (joel-oe-lacey) [reply app/controllers/reading_collections_controller.rb:6]: Swell, I'll check  on my end to be sure, but probably good to have in there regardless!

File: app/workers/reading_collections/refresh_reading_collection_worker.rb [hunk 4]
```diff
@@ -5,10 +5,12 @@ class RefreshReadingCollectionWorker
...
       reading_collections.each do |reading_collection|
         reading_collection.articles.clear
-        reading_collection.get_articles
+        articles = reading_collection.get_articles
+        reading_collection.articles << articles
+        reading_collection.update(updated_at: Time.current)
```
Comment (philjdelong) [inline app/workers/reading_collections/refresh_reading_collection_worker.rb:13]: I'm glad this worked, I was thinking along these lines as well

File: spec/workers/reading_collections/refresh_reading_collection_worker_spec.rb [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+      articles2 = ReadingCollection.find(collection2.id).articles
+      expect(articles2).to eq([article2])
+    end
+
+    it "re-populates articles for collection" do
+      user = create(:user)
+      collection1 = create(:reading_collection, name: "Coll 1",
+                                                user: user,
+                                                tag_list: %w[javascript career],
+                                                created_at: Time.current - 4.minutes,
```
Comment (philjdelong) [inline spec/workers/reading_collections/refresh_reading_collection_worker_spec.rb:31]: Niiiiiice. This is super clean

**Review by philjdelong — APPROVED** [review]

File: app/models/reading_collection.rb (hunk 2 above)
Comment (joel-oe-lacey) [reply app/models/reading_collection.rb:11]: Hah I've had the inverse problem testing the FE where they drop out of inclusion too quickly. But I just did a temporary local update on my end as I'm building things out.

**Review by joel-oe-lacey — APPROVED** [review]
Looks great!

**Pull request merged**